util/dvsim/dvsim.py

```diff
@@ -327,7 +327,8 @@ def parse_args():
                       help=('Run only up to N builds/tests at a time. '
                             'Default value 16, unless the DVSIM_MAX_PARALLEL '
                             'environment variable is set, in which case that '
-                            'is used.'))
+                            'is used. Only applicable when launching jobs '
+                            'locally.'))
```

Comment:
Can we add an error if you pass this argument and we're not launching jobs locally? Silently ignoring arguments causes all manner of confusion to people debugging. If this is absolutely impossible for some reason, at least let's print a warning.

Reply:
This is doable but a little complicated. You call `resolve_max_args()` in `parse_args()` itself, which sets `args.max_parallel` to 16. We resolve the launcher class later. It is hard to know at that point if `args.max_parallel` was set with `--max-parallel` switch. I would have to save a reference to the original `args.max_parallel` value and check it against the chosen launcher. Let me know if you have a better idea.

Reply:
Yeah, you're right: it is a bit difficult. I think it would be an improvement, though. Maybe file an issue to remind us to come back to it? (There are probably other similar cases that would be good to catch).

Reply:
Done - #6152.